File: python_backend/ranges/specialMoves/enPassant/is_enp_white_att_r.py

from ranges.specialMoves.enPassant.en_passant_white_attack_rank import en_passant_white_attack_rank

def is_enp_white_att_r(sqr):
    """return true if sqr is in the en-passant white attack rank, else false"""
    if sqr in en_passant_white_attack_rank:
        return True
    else:
        return False

if __name__ == "__main__":
    print(is_enp_white_att_r((1, 4)) == False)
    print(is_enp_white_att_r((1, 5)) == True)

File: node.js/ranges/specialMoves/enPassant/isEnpWhiteAttR.js
const  enPassantWhiteAttackRank  = require("./enPassantWhiteAttackRank");
const {xyToRf, rfToXy} = require("../../../coordType/crdCnvrt")


function isEnpWhiteAttR(sqr) {
    /*return true if sqr the.includes(is) en-passant white attack rank, else false**/
    if (enPassantWhiteAttackRank.includes(xyToRf(...sqr))) {
        return true
    }
    else {
        return false
    }
}

module.exports = isEnpWhiteAttR;

if (require.main === module) {
    
    const sampleBoardDicts = require("../../../testObjects/sampleBoardDicts")
    const printBoard = require("../../../printers/printBoard")

    var board = sampleBoardDicts["new"]
    var yes = []
    var no = []

    for (var rf of Object.keys(board)) {
        if (isEnpWhiteAttR(rfToXy(rf)))
            yes.push(rf)
        else
            no.push(rf)
    }
    printBoard(board, "green: yes, red: no", no, yes)
}
